core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -53,10 +53,15 @@ void onResponseHeaders(ResponseHeaders headers) {
         } else {
             if (req.method() == HttpMethod.HEAD) {
                 endOfStream = true;
-            } else if (status.isContentAlwaysEmpty()) {
-                setState(State.NEEDS_TRAILERS);
             } else {
-                setState(State.NEEDS_DATA_OR_TRAILERS);
+                if (!reqCtx.additionalResponseTrailers().isEmpty()) {
```

Comment:
I see, understood that even if `headers.isEndOfStream() == true`, we still always need to check `ctx.additionalResponseTrailers()`